CharacterizeTransformMatrix/CharacterizeTransformMatrix.py

```diff
@@ -0,0 +1,425 @@
+                % (rotation_angle_deg)
+            )
+            textResults.append(line)
+            if verbose:
+                print(line)
+            # expected outputs need to be filled with NaNs
+            rotation_axis = [np.NaN] * 3  # no rotation axis
+            euler_angles_xyz = [np.NaN] * 3  # no rotations...
+        else:
+            # There is rotation
+            # If you look in the direction of the rotation axis vector, positive angles mean counter-clockwise rotation.
+            # (as_rotvec() always returns non-negative angles, the rotation axis is inverted as necessary)
+            # If you point your LEFT thumb in the direction of the rotation axis, your fingers curl in the positive rotation direction
+            rotation_axis = rv / np.linalg.norm(
+                rv
+            )  # unit vector version of rotation axis
+            #
+            line = (
+                "The rotation matrix portion of this transformation rotates %0.1f degrees %s (if you look in the direction the vector points) around a vector which points to [%0.2f, %0.2f, %0.2f] (RAS)"
+                % (
+                    np.abs(rotation_angle_deg),
+                    "ccw" if rotation_angle_deg >= 0 else "cw",
+                    *rotation_axis,
+                )
+            )
+            textResults.append(line)
+            if verbose:
+                print(line)
+            #
+            # What is the best way to understand these as a sequence of rotations around coordinate axes?
+            euler_angles_xyz = r.as_euler("xyz", degrees=True)
+            # The results are the rotation angles about the positive x axis, then y, then z (in that order,
+            # and without the axes moving with the volume ("extrinsic" axes, not "intrinsic"))
+            # Same as above, positive angles mean ccw rotation if looking in the positive direction along the
+            # axis.
+            Rrot, Arot, Srot = euler_angles_xyz
+            line = (
+                "Broken down into a series of rotations around axes, the rotation matrix portion of the transformation rotates \n  %0.1f degrees %s around the positive R axis, then \n  %0.1f degrees %s around the positive A axis, then \n  %0.1f degrees %s around the positive S axis"
+                % (
+                    np.abs(Rrot),
+                    "ccw" if Rrot >= 0 else "cw",
+                    np.abs(Arot),
+                    "ccw" if Arot >= 0 else "cw",
+                    np.abs(Srot),
+                    "ccw" if Srot >= 0 else "cw",
+                )
+            )
+            textResults.append(line)
+            if verbose:
+                print(line)
+
+        #### Translation
+        translationVector = T[:3, 3]
+        line = (
+            "Lastly, this transformation translates, shifting:\n  %+0.1f mm in the R direction\n  %+0.1f mm in the A direction\n  %+0.1f mm in the S direction"
```

Comment:
I would consider removing `Lastly, ` as it does not add any information.

Reply:
Removed "Lastly" and added a final sentence about the order of the decomposed transformations